src/FakeItEasy/Creation/CreationResult.cs

```diff
@@ -9,26 +9,34 @@ namespace FakeItEasy.Creation
     internal abstract class CreationResult(int status)
     {
         private const int SuccessfulCreation = 0;
-        private const int FailedCreation = 1;
+        private const int InterimFailedCreation = 1;
+        private const int PermanentlyFailedCreation = 2;
 
-        public bool WasSuccessful => status == SuccessfulCreation;
+        private readonly int status = status;

```

Comment:
A warning I didn't understand talked about duplicating the status variable in a primary constructor for the `FailedCrerationResult` and wanted me to introduce the explicit field. Maybe this is reason enough to avoid primary constructors!

Reply:
There's always something with primary constructors that makes me not use them 🤷 

In this case I haven't been able to reproduce the warning with this minimal code:

```
internal abstract class CreationResult(int status)
{
    private const int SuccessfulCreation = 0;
    private const int InterimFailedCreation = 1;
    private const int PermanentlyFailedCreation = 2;

    public bool WasSuccessful => status == SuccessfulCreation;

    public bool IsFinal => status != InterimFailedCreation;
    
    public abstract object Result { get; }

    private class FailedCreationResult(int status, object result) : CreationResult(status)
    {
        public override object Result => result;
    }
}
```